Harvest.Web/Handlers/VerifyRoleAccessHandler.cs

```diff
@@ -30,12 +31,35 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
             {
                 return;
             }
-            
+
+
             if (await _dbContext.Permissions.AnyAsync(p => p.User.Iam == userIamId && (requirement.RoleStrings.Contains(p.Role.Name)
                 || p.Role.Name == Role.Codes.System))) // system admin should have access to all the things
             {
                 context.Succeed(requirement);
+                return;
             }
+
+            if (requirement.RoleStrings.Contains(Role.Codes.PI))
+            {
+                try
+                {
+                    //Assumes id will always be project id if PI string is used
+                    var projectId = Int32.Parse( _httpContext.HttpContext?.Request.RouteValues["id"]?.ToString());
```

Comment:
This seems like a convention that could bite us in the future. A developer placing `[Authorize(Policy = AccessCodes.PrincipalInvestigator)]` would need to remember that "id" must represent a projectId. I'm thinking like a route that is only indirectly related to a project, such as an id representing an invoice.

Reply:
@sprucely Agreed. Only other thought is to always use projectId...